docs/api/utility-process.md

```diff
@@ -0,0 +1,85 @@
+# UtilityProcess
+
+`UtilityProcess` is the equivalent of [`child_process.fork`][] API from Node.js
+but instead uses [Services API][] from Chromium to create child processes with
+Node.js and Message ports enabled.
+
+`UtilityProcess` is an [EventEmitter][event-emitter].
+
+## Class: UtilityProcess
+
+> Child process with Node.js integration spawned by Chromium.
+
+Process: [Main](../glossary.md#main-process)<br />
+
+### `new UtilityProcess(modulePath[, args][, options])`
+
+* `modulePath` string - Absolute path to the script that should run as entrypoint in the child process.
+* `args` string[] (optional) - List of string arguments that will be available as `process.argv`
+  in the child process.
+* `options` Object (optional)
+  * `env` Object - Environment key-value pairs. Default is `process.env`.
+  * `execArgv` string[] - List of string arguments passed to the executable. Default is `process.execArgv`.
+  * `serviceName` string - Name of the process that will appear in `name` property of
+    [`child-process-gone` event of `app`](app.md#event-child-process-gone).
+    Default is `node.mojom.NodeService`.
+  * `disableLibraryValidation` boolean (optional) _macOS_ - Allows launching the process using
+    Helper (Plugin).app which can code signed with entitlement that disables library validation.
+    Default is `false`.
+
+### Instance Methods
+
+#### `child.postMessage(channel, message, [transfer])`
+
+* `channel` string
+* `message` any
+* `transfer` MessagePortMain[] (optional)
+
+Send a message to the child process, optionally transferring ownership of
+zero or more [`MessagePortMain`][] objects.
+
+For example:
+
+```js
+// Main process
+const { port1, port2 } = new MessageChannelMain()
+const child = new UtilityProcess(path.join(__dirname, 'test.js'))
+child.postMessage('port', { message: 'hello' }, [port1])
+
+// Child process
+process.on('port', (e, msg) => {
```

Comment:
it feels a bit odd to emit an arbitrary event on `process`? It conflicts with all the existing events: https://nodejs.org/api/process.html#process-events and with any new ones that Node.js adds.

Reply:
Will replace with `parentPort`